Wikipedia/View Controllers/WebView/NativeSectionTitleOverlays/WMFSectionTitlesViewController.m

```diff
@@ -0,0 +1,231 @@
+        if (title) {
+            title = [title wmf_stringByRemovingHTML];
+
+            WMFTitleOverlayLabel* label = [[WMFTitleOverlayLabel alloc] init];
+            label.text      = title;
+            label.sectionId = sectionId;
+
+            [self.webView.scrollView addSubview:label];
+
+            NSLayoutConstraint* (^ constrainEqually)(NSLayoutAttribute) = ^NSLayoutConstraint*(NSLayoutAttribute attr) {
+                NSLayoutConstraint* c =
+                    [NSLayoutConstraint constraintWithItem:label
+                                                 attribute:attr
+                                                 relatedBy:NSLayoutRelationEqual
+                                                    toItem:browserView
+                                                 attribute:attr
+                                                multiplier:1.0
+                                                  constant:0];
+
+                [self.webView.scrollView addConstraint:c];
+
+                return c;
+            };
+
+            constrainEqually(NSLayoutAttributeTrailing);
+            constrainEqually(NSLayoutAttributeLeading);
+
+            NSLayoutConstraint* topConstraint = constrainEqually(NSLayoutAttributeTop);
+
+            WMFTitleOverlayModel* m = [[WMFTitleOverlayModel alloc] init];
+            m.anchor        = section[@"anchor"];
+            m.title         = title;
+            m.topConstraint = topConstraint;
+            m.yOffset       = topConstraint.constant;
+            m.label         = label;
+            m.sectionId     = sectionId;
+            [self.models addObject:m];
+        }
+    }
+
+    [self updateOverlaysPositions];
+}
+
+- (void)setupTopStaticNativeTitleOverlayLabel {
+    if (self.topStaticNativeTitleLabel) {
+        return;
+    }
+    self.topStaticNativeTitleLabel       = [[WMFTitleOverlayLabel alloc] init];
+    self.topStaticNativeTitleLabel.alpha = 0;
+    [self.webViewController.view addSubview:self.topStaticNativeTitleLabel];
+    [self.topStaticNativeTitleLabel mas_makeConstraints:^(MASConstraintMaker* make) {
+        make.left.equalTo(self.webViewController.view.mas_left);
+        make.right.equalTo(self.webViewController.view.mas_right);
+    }];
+
+    self.topStaticNativeTitleLabelTopConstraint =
+        [NSLayoutConstraint constraintWithItem:self.topStaticNativeTitleLabel
+                                     attribute:NSLayoutAttributeTop
+                                     relatedBy:NSLayoutRelationEqual
+                                        toItem:self.webViewController.topLayoutGuide
+                                     attribute:NSLayoutAttributeBottom
+                                    multiplier:1.0
+                                      constant:0];
+    [self.webViewController.view addConstraint:self.topStaticNativeTitleLabelTopConstraint];
+}
+
+- (void)updateOverlaysPositions {
+    NSArray* headingsTopOffsets = [self getSectionTitlesLocationsJSON];
```

Comment:
@bgerstle I suggested a category on UIWebView because it is presentation related… it depends on the view port. Not something MWKArticle can know without it being displayed.